user/models/base.py

```
@@ -1,231 +1,253 @@
+
+    def delete(self, *args, **kwargs):
+        if not self.is_used():
+            if self.file:
+                if default_storage.exists(self.file.name):
+                    default_storage.delete(self.file.name)
+
+        super().delete(*args, **kwargs)
+
+    def is_used(self):
+        # Check if the attachment is being used in any other models (e.g., Education, Experience, etc.)
+        from django.db.models import Q
```

Ruff (0.12.2)

`django.db.models.Q` imported but unused

Remove unused import: `django.db.models.Q`

(F401)